src/components/user/UserView.jsx

```diff
@@ -0,0 +1,95 @@
+import { useRequest } from "ahooks";
+import { toast } from "react-hot-toast";
+
+import { USER_STATUS } from "@/constants";
+import { useAdmin } from "@/models";
+
+export default function UserView({ user, run }) {
+  const { approveUser, deleteUser } = useAdmin();
+
+  const approveUserRequest = useRequest(async () => approveUser(user.id), {
+    manual: true,
+    onSuccess: () => {
+      toast.success("Employer has been approved!");
+      run();
+    },
+    onError: () => {
+      toast.error("Failed to approve Employer");
+    },
+  });
+
+  const rejectUserRequest = useRequest(async () => deleteUser(user.id), {
+    manual: true,
+    onSuccess: () => {
+      toast.success("User has been removed!");
+      run();
+    },
+    onError: () => {
+      toast.error("Failed to remove user");
+    },
+  });
+
+  const reject = async () => {
+    rejectUserRequest.run(user.id);
+  };
+
+  const approve = async () => {
+    approveUserRequest.run(user.id);
+  };
+
+  const stylePhoneNumber = (number) => {
+    if (number.length !== 10) {
+      return number;
+    } else {
+      return `${number.slice(0, 3)}-${number.slice(3, 6)}-${number.slice(6, 11)}`;
+    }
+  };
```

Comment:
could you help me add an todo
`todo: make the styling to other places that show phone number`